.clang-format

```diff
@@ -1,101 +1,199 @@
+ - list_for_each_entry_reverse_safe
+ - list_for_each_entry_safe
+ - s_list_for_each
+ - s_list_for_each_entry
+ - s_list_for_each_entry_safe
+ - s_list_for_each_safe
+WhitespaceSensitiveMacros:
+BinPackArguments: true
 BinPackParameters: false
 BraceWrapping:
-  AfterClass:      true
-  AfterControlStatement: false
-  AfterEnum:       false
-  AfterFunction:   true
-  AfterNamespace:  false
+  AfterCaseLabel: false
+  AfterClass: false
+  AfterControlStatement: MultiLine
+  AfterEnum: false
+  AfterFunction: true
+  AfterNamespace: false
   AfterObjCDeclaration: false
-  AfterStruct:     false
-  AfterUnion:      false
-  BeforeCatch:     false
-  BeforeElse:      false
-  IndentBraces:    false
+  AfterStruct: false
+  AfterUnion: false
+  AfterExternBlock: false
+  BeforeCatch: false
+  BeforeElse: false
+  BeforeLambdaBody: false
+  BeforeWhile: false
+  IndentBraces: false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
 BreakBeforeBinaryOperators: None
+BreakBeforeConceptDeclarations: true
 BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: AfterColon
 BreakBeforeTernaryOperators: true
-BreakConstructorInitializersBeforeComma: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: AfterColon
 BreakAfterJavaFieldAnnotations: false
-BreakStringLiterals: true
-ColumnLimit:     100
-CommentPragmas:  '^ IWYU pragma:'
+BreakStringLiterals: false
+ColumnLimit: 100
+CommentPragmas: '^ IWYU pragma:'
+CompactNamespaces: false
 ConstructorInitializerAllOnOneLineOrOnePerLine: false
-ConstructorInitializerIndentWidth: 2
+ConstructorInitializerIndentWidth: 4
 ContinuationIndentWidth: 4
 Cpp11BracedListStyle: false
-DerivePointerAlignment: true
-DisableFormat:   false
+DeriveLineEnding: true
+DerivePointerAlignment: false
+DisableFormat: false
+EmptyLineBeforeAccessModifier: Always
 ExperimentalAutoDetectBinPacking: false
-ForEachMacros:
-  - list_for_each
-  - list_for_each_entry
-  - list_for_each_entry_reverse
-  - list_for_each_entry_safe
-  - foreach
-  - Q_FOREACH
-  - BOOST_FOREACH
+FixNamespaceComments: true
+IncludeBlocks: Regroup
 IncludeCategories:
-  - Regex:           '^"(llvm|llvm-c|clang|clang-c)/'
-    Priority:        2
-  - Regex:           '^(<|"(gtest|isl|json)/)'
-    Priority:        3
-  - Regex:           '.*'
-    Priority:        1
-IncludeIsMainRegex: '$'
-IndentCaseLabels: true
-IndentWidth:     4
+ - Regex:           '^[<"]hse/.+/'
+   Priority:        11
+ - Regex:           '^[<"]hse/'
+   Priority:        10
+ - Regex:           '^[<"]build_config.h'
+   Priority:        1
+ - Regex:           '^"'
+   Priority:        12
+ - Regex:           '^<.*/'
+   Priority:        3
+ - Regex:           '^<'
+   Priority:        2
+ - Regex:           '.'
+   Priority:        100
+# Seems to be no way to disable the "IncludeIsMain" settings,
+# so we set them to patterns we don't expect to see.
+IncludeIsMainRegex: 'XXX'
```

Comment:
What is the point of including these if we aren't using them?

Reply:
`IncludeIsMainSourceRegex ` has a default behavior that doesn't match our include conventions.  Documentation doesn't specify a way to disable the default behavior.  I set them both to something that should never match just to be safe.

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includeismainsourceregex

Reply:
So it seems like this has something to do with if you have a source file named "my_xxx.c", then it would do something special with "my_xx.h"? I didn't really gather what was going on from the description. Really needs an example in the docs.

Reply:
You nailed it.  It will put "my_xx.h" as priority 0 (first include file).

Reply:
I did not.  But I'm happy to include that in the post reformat cleanup of .clang-format when it will be cinch to know if the changes impact the format.

Reply:
removed one XXX but had to keep the other.